src/store/subjects.js

```diff
@@ -1,44 +1,41 @@
+import { ADD_SUBJECT, TOGGLE_SUBJECT, REMOVE_SUBJECT, UPDATE_SUBJECT } from '../constants/ActionTypes'
+
 const subjects = (state = [], action) => {
   switch (action.type) {
-    case 'ADD_SUBJECT':
+    case ADD_SUBJECT:
       return [
         ...state,
         {
-          id: action.id,
+          id: action.payload.id,
           name: '',
-          type: action.subjectsType,
+          type: action.payload.subjectsType,
           elementaryScore: 0,
           advancedScore: 0,
           active: true
         }
       ]
-    case 'TOGGLE_SUBJECT':
+    case TOGGLE_SUBJECT:
       return state.map(
-        subject => (subject.id === action.id) ? {
-          id: subject.id,
-          name: subject.name,
-          type: subject.type,
-          elementaryScore: subject.elementaryScore,
-          advancedScore: subject.advancedScore,
+        subject => (subject.id === action.payload.id) ? {
+          ...subject,
           active: !subject.active
         }
         : subject
       )
-    case 'REMOVE_SUBJECT':
+    case REMOVE_SUBJECT:
       return state.filter(
-        subject => subject.id !== action.id
+        subject => subject.id !== action.payload.id
       )
-    case 'UPDATE_SUBJECT':
+    case UPDATE_SUBJECT:
       return state.map(
-        subject => (subject.id === action.id) ? {
-          id: action.id,
-          name: action.name,
-          type: subject.type,
-          elementaryScore: action.elementaryScore,
-          advancedScore: action.advancedScore,
-          active: subject.active
-        }
-        : subject
+        subject => (subject.id === action.payload.id)
+          ? {
+            ...subject,
+            name: action.payload.name,
```

Comment:
I haven't destructured only beceause I wanted the code to be more human-readable, so I know what's changing there.